hr_expense_product_policy/models/hr_expense.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# © 2016 OpenSynergy Indonesia
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models
+from openerp.tools import ustr
+
+
+class HrExpense(models.Model):
+    _inherit = "hr.expense.expense"
+
+    @api.depends(
+        "employee_id",
+        "employee_id.required_expense_product")
+    def _compute_required_expense_product(self):
+        for expense in self:
+            expense.required_expense_product = \
+                expense.employee_id.required_expense_product
+
+    @api.depends(
+        "employee_id",
+        "employee_id.all_allowed_expense_product_ids")
+    def _compute_all_allowed_product_ids(self):
+        for expense in self:
+            expense.all_allowed_expense_product_ids = \
+                expense.employee_id.all_allowed_expense_product_ids.mapped(
+                    "id")
+
+    required_expense_product = fields.Boolean(
+        string="Required Expense Product",
+        compute="_compute_required_expense_product",
+        store=True,
+        )
+    all_allowed_expense_product_ids = fields.Many2many(
+        string="All Allowed Expense Product",
+        comodel_name="product.product",
+        relation="rel_all_allowed_product_2_expense",
+        column1="expense_id",
+        column2="product_id",
+        compute="_compute_all_allowed_product_ids",
+        store=False,
+    )
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        """Inject the domain here to avoid conflicts with other modules.
```

Comment:
I want to avoid conflict if other module add/modify domain on product field. Do you have concern that i should consider when overriding fields_view_get?

I got the example from https://github.com/odoomrp/odoomrp-wip/tree/8.0/sale_order_allowed_product and https://github.com/odoomrp/odoomrp-wip/tree/8.0/purchase_order_allowed_product.